src/ComputeSharp.D2D1/Shaders/Interop/Helpers/Rectangle64.cs

```diff
@@ -0,0 +1,159 @@
+    }
+
+    /// <summary>
+    /// Gets the bottom coordinate of the current rectangle.
+    /// </summary>
+    public readonly long Bottom
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get => this.y + this.height;
+    }
+
+    /// <summary>
+    /// Gets a <see cref="Rectangle64"/> instance representing an empty rectangle.
+    /// </summary>
+    /// <remarks>The returned instance is such that computing the union of it with any othe rectangle yields the other rectangle.</remarks>
+    public static Rectangle64 Empty => new(long.MaxValue, long.MaxValue, long.MinValue, long.MinValue);
+
+    /// <summary>
+    /// Updates the current rectangle to be the union of itself and a given input rectangle.
+    /// </summary>
+    /// <param name="rectangle">The input <see cref="Rectangle64"/> instance to compute the union with.</param>
+    public void Union(Rectangle64 rectangle)
+    {
+        long left = Math.Min(this.Left, rectangle.Left);
+        long top = Math.Min(this.Top, rectangle.Top);
+        long right = Math.Max(this.Right, rectangle.Right);
+        long bottom = Math.Max(this.Bottom, rectangle.Bottom);
+
+        this.x = left;
+        this.y = top;
+        this.width = right - left;
+        this.height = bottom - top;
+    }
+
+    /// <summary>
+    /// Updates the current rectangle to be the intersection of itself and a given input rectangle.
+    /// </summary>
+    /// <param name="rectangle">The input <see cref="Rectangle64"/> instance to compute the intersection with.</param>
+    public void Intersect(Rectangle64 rectangle)
+    {
+        long maxLeft = Math.Max(this.x, rectangle.x);
+        long maxTop = Math.Max(this.y, rectangle.y);
+        long width = Math.Max(Math.Min(this.Right, rectangle.Right) - maxLeft, 0);
+        long height = Math.Max(Math.Min(this.Bottom, rectangle.Bottom) - maxTop, 0);
+
+        this.x = maxLeft;
+        this.y = maxTop;
+        this.width = width;
+        this.height = height;
+    }
+
+    /// <summary>
+    /// Inflates the current rectangle by a specified amount.
+    /// </summary>
+    /// <param name="left">The amount fo use to inflate the left edge.</param>
+    /// <param name="top"></param>
+    /// <param name="right"></param>
+    /// <param name="bottom"></param>
+    public void Inflate(long left, long top, long right, long bottom)
+    {
+        this.x -= left;
+        this.y -= top;
+        this.width += left + right;
+        this.height += top + bottom;
+    }
+
+    /// <summary>
+    /// Creates a new <see cref="Rectangle"/> from the current <see cref="Rectangle64"/> instance, clamping to a D2D1 logically infinite rectangle.
+    /// </summary>
+    /// <returns>The <see cref="Rectangle"/> equivalent of the current instance.</returns>
+    public Rectangle ToRectangleWithD2D1LogicallyInfiniteClamping()
+    {
+        Intersect(new Rectangle64(int.MinValue / 2, int.MinValue / 2, int.MaxValue, int.MaxValue));
```

Comment:
definitely recommend stashing this into a `static readonly Rectangle LogicallyInfinite` somewhere. Way easier to read and re-use. I was being lazy in my own codebase and didn't do it that way for awhile and it got obnoxious.

Reply:
Eh, for now this is the only use, so not worth a property. I'll add one in case more uses pop up 🙂